scripts/image

```diff
@@ -82,487 +82,479 @@ if [ -n "$IMAGE_SUFFIX" ]; then
+
+  # create image files if requested
+  if [[ ( "$1" = "amlpkg" || "$1" = "noobs" || "$1" = "mkimage" ) && -n "$BOOTLOADER" ]]; then
+    # projects can set KERNEL_NAME (kernel.img)
+    if [ -z "$KERNEL_NAME" ] ; then
+      KERNEL_NAME="KERNEL"
+    fi
+
+    UBOOT_SYSTEM=""
+    if [ "$PROJECT" == "imx6" ]; then
+      if [ -n "$SYSTEM" ]; then
+        UBOOT_SYSTEM="$SYSTEM"
+      else
+        UBOOT_SYSTEM="cuboxi"
+      fi
+    fi
+
+    # variables used in image script must be passed
+    env \
+      PATH="$PATH:/sbin" \
+      ROOT="$ROOT" \
+      TOOLCHAIN="$TOOLCHAIN" \
+      PROJECT_DIR="$PROJECT_DIR" \
+      PROJECT="$PROJECT" \
+      DISTRO="$DISTRO" \
+      TARGET_IMG="$TARGET_IMG" \
+      IMAGE_NAME="$IMAGE_NAME" \
+      BOOTLOADER="$BOOTLOADER" \
+      KERNEL_NAME="$KERNEL_NAME" \
+      RELEASE_DIR=$RELEASE_DIR \
+      UUID_SYSTEM="$(uuidgen)" \
+      UUID_STORAGE="$(uuidgen)" \
+      UBOOT_SYSTEM="$UBOOT_SYSTEM" \
+      EXTRA_CMDLINE="$EXTRA_CMDLINE" \
+      SYSTEM_SIZE="$SYSTEM_SIZE" \
+      SYSTEM_PART_START="$SYSTEM_PART_START" \
+      $SCRIPTS/mkimage
+  fi
 
-    # cleanup
-      rm -rf $RELEASE_DIR
+  # cleanup release dir
+  rm -rf $RELEASE_DIR
+
+  # create WeTek Play (Amlogic) ZIP update and auto-install packages if requested
+  if [ "$1" = "amlpkg" ]; then
+    echo "Creating Amlogic ZIP update package"
+
+    INSTALL_SRC_DIR="$PROJECT_DIR/$PROJECT/install"
+    AML_PKG_DIR="$RELEASE_DIR/ampl-pkg"
+
+    # create package directory
+    mkdir -p "$AML_PKG_DIR"
+
+    # copy system and kernel images
+    mkdir -p "$AML_PKG_DIR/system"
+    cp $TARGET_IMG/$IMAGE_NAME.system $AML_PKG_DIR/system/SYSTEM
+    cp $TARGET_IMG/$IMAGE_NAME.kernel $AML_PKG_DIR/KERNEL
+
+    # copy update-binary and updater-script
+    META_INF_DIR="$AML_PKG_DIR/META-INF/com/google/android"
+    mkdir -p "$META_INF_DIR"
+    cp $INSTALL_SRC_DIR/update-binary $META_INF_DIR
+    cp $INSTALL_SRC_DIR/updater-script $META_INF_DIR
+
+    # copy other files if any
+    if [ -d "$INSTALL_SRC_DIR/files" ]; then
+      cp -PR $INSTALL_SRC_DIR/files/* $AML_PKG_DIR
+    fi
+
+    # copy device tree image if any
+    if [ -f "$INSTALL/usr/share/bootloader/dtb.img" ]; then
+      cp "$INSTALL/usr/share/bootloader/dtb.img" $AML_PKG_DIR/dtb.img
+    fi
+
+    # create the update package
+    pushd "$AML_PKG_DIR" > /dev/null
+    zip -rq update.zip *
+
+    # sign the update package
+    echo "Signing the update package"
+    mkdir -p sign
+    SIGNAPK_DIR="$ROOT/tools/signapk"
+    java -Xmx1024m -jar $SIGNAPK_DIR/signapk.jar -w $SIGNAPK_DIR/testkey.x509.pem $SIGNAPK_DIR/testkey.pk8 update.zip sign/$IMAGE_NAME-update.zip
+
+    # create the auto-install package
+    echo "Creating Amlogic ZIP auto-install package"
+    pushd sign > /dev/null
+    echo --update_package=/sdcard/$IMAGE_NAME-update.zip > factory_update_param.aml        
+    if [ -f "$INSTALL_SRC_DIR/files/recovery.img" ]; then
+      cp $INSTALL_SRC_DIR/files/recovery.img .
```

Comment:
I also changed the logic here, because the lines 435 to 453 where inside the `if [ -f "$INSTALL_SRC_DIR/files/recovery.img" ]` block, which is imho not correct. Forgot to mention it in the pr.